src/domain.cc

```diff
@@ -551,166 +492,71 @@ NLV_WORKER_EXECUTE(Domain, GetAutostart)
-
-  data_ = result;
+  RunMethod<MethodReturn<double>>(info, virDomainGetMaxMemory);
 }
 
-NLV_WORKER_METHOD_NO_ARGS(Domain, GetMaxVcpus)
-NLV_WORKER_EXECUTE(Domain, GetMaxVcpus)
+NAN_METHOD(Domain::GetMaxVcpus)
 {
-  NLV_WORKER_ASSERT_DOMAIN();
-  int result = virDomainGetMaxVcpus(Handle());
-  if (result == -1) {
-    SetVirError(virSaveLastError());
-    return;
-  }
-
-  data_ = result;
+  RunMethod<MethodReturnInt>(info, virDomainGetMaxVcpus);
 }
 
-NLV_WORKER_METHOD_NO_ARGS(Domain, IsActive)
-NLV_WORKER_EXECUTE(Domain, IsActive)
+NAN_METHOD(Domain::IsActive)
 {
-  NLV_WORKER_ASSERT_DOMAIN();
-  int result = virDomainIsActive(Handle());
-  if (result == -1) {
-    SetVirError(virSaveLastError());
-    return;
-  }
-
-  data_ = static_cast<bool>(result);
+  RunMethod<MethodReturnBool>(info, virDomainIsActive);
 }
 
-NLV_WORKER_METHOD_NO_ARGS(Domain, IsPersistent)
-NLV_WORKER_EXECUTE(Domain, IsPersistent)
+NAN_METHOD(Domain::IsPersistent)
 {
-  NLV_WORKER_ASSERT_DOMAIN();
-  int result = virDomainIsPersistent(Handle());
-  if (result == -1) {
-    SetVirError(virSaveLastError());
-    return;
-  }
-
-  data_ = static_cast<bool>(result);
+  RunMethod<MethodReturnBool>(info, virDomainIsPersistent);
 }
 
-NLV_WORKER_METHOD_NO_ARGS(Domain, IsUpdated)
-NLV_WORKER_EXECUTE(Domain, IsUpdated)
+NAN_METHOD(Domain::IsUpdated)
 {
-  NLV_WORKER_ASSERT_DOMAIN();
-  int result = virDomainIsUpdated(Handle());
-  if (result == -1) {
-    SetVirError(virSaveLastError());
-    return;
-  }
-
-  data_ = static_cast<bool>(result);
+  RunMethod<MethodReturnBool>(info, virDomainIsUpdated);
 }
 
-
-NLV_WORKER_METHOD_NO_ARGS(Domain, Start)
-NLV_WORKER_EXECUTE(Domain, Start)
+NAN_METHOD(Domain::Start)
 {
-  NLV_WORKER_ASSERT_DOMAIN();
-  int result = virDomainCreate(Handle());
-  if (result == -1) {
-    SetVirError(virSaveLastError());
-    return;
-  }
-
-  data_ = true;
+  RunMethod(info, virDomainCreate);
 }
 
-NLV_WORKER_METHOD_NO_ARGS(Domain, Reboot)
-NLV_WORKER_EXECUTE(Domain, Reboot)
+NAN_METHOD(Domain::Reboot)
 {
-  NLV_WORKER_ASSERT_DOMAIN();
-  unsigned long flags = 0;
-  int result = virDomainReboot(Handle(), flags);
-  if (result == -1) {
-    SetVirError(virSaveLastError());
-    return;
-  }
-
-  data_ = true;
+  RunMethod(info, virDomainReboot, GetFlags(info[0]));
 }
 
-NLV_WORKER_METHOD_NO_ARGS(Domain, Reset)
-NLV_WORKER_EXECUTE(Domain, Reset)
+NAN_METHOD(Domain::Reset)
 {
-  NLV_WORKER_ASSERT_DOMAIN();
-  unsigned long flags = 0;
-  int result = virDomainReset(Handle(), flags);
-  if (result == -1) {
-    SetVirError(virSaveLastError());
-    return;
-  }
-
-  data_ = true;
+  RunMethod(info, virDomainReset, GetFlags(info[0]));
 }
 
-NLV_WORKER_METHOD_NO_ARGS(Domain, Shutdown)
-NLV_WORKER_EXECUTE(Domain, Shutdown)
+NAN_METHOD(Domain::Shutdown)
 {
-  NLV_WORKER_ASSERT_DOMAIN();
-  int result = virDomainShutdown(Handle());
-  if (result == -1) {
-    SetVirError(virSaveLastError());
-    return;
-  }
-
-  data_ = true;
+#if LIBVIR_CHECK_VERSION(0,9,10)
```

Comment:
you should check what actual version of libvirt this applies to, I had to fix the last feature you wrapped with this (I think related to some sort of block feature) because the version wasn't correct. Unfortunately the process for that is basically tromping through the libvirt changelog and figuring it out manually